jvm/ca.ualberta.maple.swan.drivers/src/resources/models.swirl

```diff
@@ -246,6 +744,143 @@ bb0(%0 : $`*τ_0_1`, %1 : $`*τ_0_0`, %2 : $`τ_0_1`, %3 : $`Dictionary<τ_0_0,
+bb0(%0 : $`*Zip2Sequence<Array<τ_0_0>, Array<τ_0_1>>`, %1 : $`(@out String, @error Error)`, %2 : $`*Dictionary<τ_0_0, τ_0_1>`):
+  zipSeq = pointer_read %0, $`Zip2Sequence<Array<τ_0_0>, Array<τ_0_1>>`
+  new_dict = new $`Dictionary<τ_0_0, τ_0_1>`
+  Arr = pointer_read zipSeq, $`Array<τ_0_1>`
+  value = pointer_read Arr, $`τ_0_1`
+  dict = pointer_read %2, $`Dictionary<τ_0_0, τ_0_1>`
+  pointer_write value to new_dict
+  pointer_write [weak] new_dict to dict
+  %ret = new $`@error Error`
+  return %ret
+}
+
+// %0 -> bool type, %1 -> dictionary
+// Returns a new dictionary containing the key-value pairs of the dictionary
+// that satisfy the given predicate.
+func [model] @`Swift.Dictionary.filter(((key: A, value: B)) throws -> Swift.Bool) throws -> [A : B]` : $`(@owned Dictionary<τ_0_0, τ_0_1>, @error Error)` {
+bb0(%0 : $`(Bool, @error Error)`, %1 : $`Dictionary<τ_0_0, τ_0_1>`):
+  return %1
+}
+
+// %0 -> dictionary that needs to be read a value
+// get the value from dictionary
+func [model] @`Swift.Dictionary.values.getter : [A : B].Values` : $`@owned Dictionary<τ_0_0, τ_0_1>.Values` {
+bb0(%0 : $`Dictionary<τ_0_0, τ_0_1>`):
+  elem = new $`@owned Dictionary<τ_0_0, τ_0_1>.Values`
+  val = pointer_read %0, $`τ_0_1`
+  pointer_write val to elem
+  return elem
+}
+
+// %1 -> dictionary
+// Returns a new dictionary containing the keys of this dictionary with
+// the values transformed by the given closure.
+func [model] @`Swift.Dictionary.mapValues<A>((B) throws -> A1) throws -> [A : A1]` : $`(@owned Dictionary<τ_0_0, τ_1_0>, @error Error)` {
+bb0(%0 : $`(@out String, @error Error)`, %1 : $`@owned Dictionary<τ_0_0, τ_0_1>`):
+  dict = new $`(@owned Dictionary<τ_0_0, τ_1_0>, @error Error)`
+  val = pointer_read %1, $`τ_0_1`
+  pointer_write val to dict
+  return dict
+}
+
+// %0 -> converted String
+// convert Uppercase letter to lowercase letter
+func [model] @`Swift.String.lowercased() -> Swift.String` : $`@owned String` {
+bb0(%0 : $`@guaranteed String`):
+  return %0
+}
+
+// %0 -> String, %1 -> String
+// two Strings for comparison
+// compare the values
+func [model] @`static (extension in Swift):Swift.Comparable.> infix(A, A) -> Swift.Bool` : $`Bool` {
+bb0(%0 : $`*τ_0_0`, %1 : $`*τ_0_0`, %2 : $`@thick τ_0_0.Type`):
+  %ret = new $`Bool`
+  return %ret
+}
+
+// %1 -> dictionary
+// Returns a new dictionary containing only the key-value pairs
+// that have non-nil values as the result of transformation by the given closure.
+func [model] @`Swift.Dictionary.compactMapValues<A>((B) throws -> A1?) throws -> [A : A1]` : $`(@owned Dictionary<τ_0_0, τ_1_0>, @error Error)` {
+bb0(%0 : $`(@out Optional<String>, @error Error)`, %1 : $`@owned Dictionary<τ_0_0, τ_0_1>`):
+  dict = new $`(@owned Dictionary<τ_0_0, τ_1_0>, @error Error)`
+  val = pointer_read %1, $`τ_0_1`
+  key = pointer_read %1, $`τ_0_0`
```

Comment:
We don't need to worry about the `key`. Also, the two pointer reads are semantically the same, anyway.